src/main/java/org/jitsi/videobridge/cc/VideoAllocationPolicy.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc;
+
+import org.jitsi.videobridge.*;
+import org.json.simple.*;
+
+import java.util.*;
+
+public class VideoAllocationPolicy
+    extends VideoConstraints
```

Comment:
I'm confused by this extending `VideoConstraints`, as I think the two things are very different.  The way I see it is:

1) Receivers create `VideoConstraints` instances that express how they would, ideally, view other senders.
2) Senders combine all the `VideoConstraints` that have been set for them to determine a lowest common denominator and determine if they can suspend certain layers (e.g. if the highest expressed constraint is 360p, it can stop sending 720p).
3. Totally separately, the `BitrateController` for an endpoint, when deciding what to forward, uses a `VideoAllocationPolicy` to determine how to allocate the bandwidth available across the streams.  For example, say a receiver was in tile view and had expressed they'd, ideally, see every sender in 720p, but their bandwidth is insufficient to handle this.  A `PreferMoreVideo` `VideoAllocationPolicy` would tradeoff quality to forward more streams (i.e. forward the 360p layer of all senders).  A `PreferQuality` `VideoAllocationPolicy` would trade off more streams to send better quality (i.e. maximize the forwarded quality of each sender one by one and, for some senders, not end up forwarding any video at all).

Reply:
@bbaldino would a reasonable compromise at this point be to separate the `VideoAllocationPolicy` from the `VideoConstraints` but leave the fields as they are now (i.e. `VideoAllocationPolicy { preferredFps, preferredHeight }`

Reply:
Actually I may be able to tweak the bitrate allocator to suit your suggestion. I'll work on this tomorrow.